docs/source/quick-start.rst

```diff
@@ -200,3 +200,151 @@ read-write and read-only pool instances:
+ * `on_push_ctx` - result of the `on_push` work can be written to this variable.
+
+Below is an example of the proposed API with method :meth:`~tarantool.Connection.call` 
+and :meth:`~tarantool.Connection.insert`. In the described example, before the end 
+of the :meth:`~tarantool.Connection.call` and :meth:`~tarantool.Connection.insert`, 
+out-of-band messages are processed via specified callback.
+
+In the example below, two shells are used, in the first we will configure the server:
+
+.. code-block:: lua
+
+    fiber = require('fiber')
+    box.cfg({listen = 3301})
+    box.schema.user.grant(
+        'guest',
+        'read,write,execute',
+        'universe'
+    )
+    function server_function()
+        x = {0,0}
+        while x[1] < 3 do
+            x[1] = x[1] + 1
+            fiber.sleep(1)
+            box.session.push(x)
+        end
+        fiber.sleep(1)
+        return x
+    end
+
+In the second shell, we will execute a :meth:`~tarantool.Connection.call` 
+with receiving out-of-band messages from the server:
+
+.. code-block:: python
+
+    import tarantool
+
+    def callback(data, on_push_ctx=[]):
+        print('run callback with data: ', data)
+        data[0][1] = data[0][1] + 1
+        on_push_ctx.append(data)
+
+    callback_res = []
+
+    conn = tarantool.connect(port=3301)
+    res = conn.call(
+        'server_function',
+        on_push=callback, 
+        on_push_ctx=callback_res
+    )
+
+    # receiving out-of-band messages, 
+    # the conn.call is not finished yet.
+
+    >>> run callback with data:  [[1, 0]]
+    >>> run callback with data:  [[2, 0]]
+    >>> run callback with data:  [[3, 0]]
+
+    # the conn.call is finished now.
+
+    print(res)
+    >>> [3, 0]
+
+    print(callback_res)
+    >>> [[[1, 1]], [[2, 1]], [[3, 1]]]
+
+Let's go back to the first shell with the server and 
+create a space and a trigger for it:
+
+.. code-block:: lua
+
+    box.schema.create_space(
+     'tester', {
+      format = {
+        {name = 'id', type = 'unsigned'},
+        {name = 'name', type = 'string'},
+      }
+    })
+    box.space.tester:create_index(
+     'primary_index', {
+      parts = {
+        {field = 1, type = 'unsigned'},
+      }
+    })
+    function on_replace_callback()
+        x = {0,0}
+        while x[1] < 300 do
+            x[1] = x[1] + 100
+            box.session.push(x)
+        end
+        return x
+    end
+    box.space.tester:on_replace(
+        on_replace_callback
+    )
+
+Now, in the second shell, we will execute an :meth:`~tarantool.ConnectionPool.insert` 
+with out-of-band message processing:
+
+.. code-block:: python
+
+    callback_res = []
+
+    conn_pool = tarantool.ConnectionPool(
+            [{'host':'localhost', 'port':3301}],
+            user='guest')
+
+    res = conn_pool.insert(
+        'tester',
+        (1, 'Mike'),
+        mode=tarantool.Mode.PREFER_RO,
```

Comment:
`PREFER_RO` is a bit specific mode -- the only case when it's useful is when you want to read from replica so there won't be any excessive load for read-write instances which are busy with writing. I think it's better to use `ANY` for calls if there is no any reason to do otherwise and default mode where there is one.

Reply:
Done